src/read/mod.rs

```diff
@@ -168,14 +162,14 @@ where
         // in all cases, using a pointer we read *after* updating our epoch is safe.
 
         // so, update our epoch tracker.
-        let epoch = self.my_epoch.fetch_add(1, atomic::Ordering::Relaxed);
-        self.epoch.store(epoch + 1, atomic::Ordering::Release);
+        let epoch = self.my_epoch.fetch_add(1, Ordering::Relaxed);
+        // self.epoch.store(epoch + 1, Ordering::Release);
```

Comment:
This is pretty interesting. This logic just might be too complicated for the current `loom` capabilities. If I uncomment this line and run the test in `tests/loom` (which I _think_ is right and I _think_ is about as simple as it's going to get?) I get:
> thread 'loom_tests::evmap_read_while_remove' panicked at 'Model exeeded maximum number of branches. This is often caused by an algorithm requiring the processor to make progress,
e.g. spin locks.'

If I comment this out, the test passes. We might need to either simplify the logic or start getting some PRs into `loom` to beef it up? Currently it looks like the thread max is handled by a `u16` and maybe we can just bump that up? 🤷 Interested in your thoughts! Would also be free sometime to jump on a call to work through this in a more synchronous manner depending upon both our lives

Reply:
And obviously this line isn’t special - just showing that we might be at the cusp of complexity

Reply:
Or maybe I just did a bad job increasing the default max with env vars

Reply:
Hmm, that's interesting. We _are_ spinning in the writer waiting for the reader to finish its read, that is certainly true. I think maybe the trick here is going to be to have
https://github.com/jonhoo/rust-evmap/blob/34b5306be23063579dcf5a7aefba93af39eb4ade/src/write.rs#L204

Always call `yield_now` rather than spin when running under `loom`. That might fix it.

Reply:
I was wondering about this because we don't call `refresh` after the reader so I didn't think this was possible. I was only looking at `remove_entry` which didn't do anything interesting.

Is it that `loom` has scheduled the `refresh` at the beginning of the test to be after when the reader reads? I assumed it wouldn't move around "sequential" operations (which is why I also had the `w.remove_entry` in a thread - do I just need no threads and it'll reorder everything anyway?). Maybe I need to read their docs more slowly and/or do a mini deep dive into the code to figure out how it works.

Reply:
Hmm, no, you're right, since there isn't a refresh after the operation, that shouldn't be spinning, since as you say spawning a thread is a synchronizing operation. I still think we'll need the change I outlined above for other tests, but not for this one.

I'd try jumping in the loom channel and see if anyone can shed some light there.